src/wallet/wallet.cpp

```diff
@@ -1814,10 +1814,7 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend,
                     dPriority += (double)nCredit * age;
                 }
 
-                CAmount nChange = nValueIn - nValue;
-                if (nSubtractFeeFromAmount == 0)
-                    nChange -= nFeeRet;
-
+                const CAmount nChange = nValueIn - nValueToSelect;
```

Comment:
Before

    nChange = nValueIn - nValue - nSubtractFeeFromAmount?0:nFeeRet 

After:

    nChange = nValueIn - nValueToSelect 
            = nValueIn - (nValue + nSubtractFeeFromAmount?0:nFeeRet) 
            = nValueIn - nValue - nSubtractFeeFromAmount?0:nFeeRet

Looks good to me!
Looks good to me!